Toxcore does not build with cmake on OpenBSD.

I've taken a look into toxcore/ccompat.h and this is what worries me:
```
#ifdef _WIN32
#include <malloc.h>
#elif defined(__FreeBSD__)
#include <stdlib.h>
#if !defined(alloca) && defined(__GNUC__)
#define alloca __builtin_alloca
#endif
#else
#include <alloca.h>
#endif
```
As far as I know <alloca.h> is a Linux-specific header (and perhaps also MacOS-specific header). That means toxcore works only at Windows, Linux and FreeBSD...
@iphydf Do you know that alloca(3) and malloc(3) are supposed to be at <stdlib.h> at any OS, including Windows and Linux? So this should work not only for FreeBSD:
```
-#ifdef _WIN32
-#include <malloc.h>
-#elif defined(__FreeBSD__)
 #include <stdlib.h>
 #if !defined(alloca) && defined(__GNUC__)
 #define alloca __builtin_alloca
 #endif
-#else
-#include <alloca.h>
-#endif
```
But quickly patching that doesn't help, I get a spam of undefined references: https://pastebin.com/raw/NbrasTWR
What have you done? You made toxcore completely non portable!

@iphydf 

First, I don't have <alloca.h> header:
<pre>
test.c:1:10: fatal error: 'alloca.h' file not found
#include <alloca.h>
         ^~~~~~~~~~
1 error generated.
</pre>

Second, leaving it as is does not help to build (it didn't start building toxcore/ccompat.h, but at release 0.2.0 it returned error because toxcore/ccompat.h tryid to include <alloca.h>), still a flood of undefined references: https://pastebin.com/raw/8g9iyw2i

Third, according to man pages at Linux, FreeBSD, NetBSD and OpenBSD alloca() is terrible.

Fourth, it seems that including <stdlib.h> at Linux will work if you don't use -ansi and -std=c* with gcc https://man.openbsd.org/Linux-4.13/alloca#Notes_on_the_GNU_version

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -62,6 +62,18 @@ if(APPLE)
   include(MacRpath)
 endif()
 
+if(UNIX)
+    if(CMAKE_SYSTEM_NAME MATCHES ".*Linux")
+        set(LINUX TRUE)
+    elseif(CMAKE_SYSTEM_NAME MATCHES "kOpenBSD.*|OpenBSD.*")
+        set(OPENBSD TRUE)
+    elseif(CMAKE_SYSTEM_NAME MATCHES "kNetBSD.*|NetBSD.*")
+        set(NETBSD TRUE)
+    elseif(CMAKE_SYSTEM_NAME MATCHES "kFreeBSD.*|FreeBSD")
+        set(FREEBSD TRUE)
+    endif()
+endif()
+
 enable_testing()
 
 set(CMAKE_MACOSX_RPATH ON)
@@ -184,10 +196,10 @@ if(ASAN)
 else()
   # Forbid undefined symbols in shared libraries. This is incompatible with
   # asan, so it's in the else branch here.
-  if(APPLE)
-    add_dllflag("-undefined error")
-  else()
+  if(LINUX)
     add_dllflag("-Wl,-z,defs")
+  else()
+    add_dllflag("-undefined error")
   endif()
 endif()
 

--- a/toxcore/ccompat.h
+++ b/toxcore/ccompat.h
@@ -22,13 +22,13 @@
 // Emulation using alloca.
 #ifdef _WIN32
 #include <malloc.h>
-#elif defined(__FreeBSD__)
+#elif defined(__linux__)
+#include <alloca.h>
+#else
 #include <stdlib.h>
 #if !defined(alloca) && defined(__GNUC__)
 #define alloca __builtin_alloca
 #endif
-#else
-#include <alloca.h>
 #endif
 
 #define VLA(type, name, size)                           \

